### ISSUE TITLE
DtlsTransport issue

### ISSUE THREAD

**Issue by pezalot** [paullouisageneau/libdatachannel#907]
Hi, I have strange iisue according to DtlsTransport::Init() method. First 10x for the effort you made to create this project appriciate!!!. Second my build and compile procedure is with make: make USE_GNUTLS=0 USE_NICE=0 NO_MEDIA=1 NO_WEBSOCKET=1 -j22 and everythink is fine. Third my configuration for peer connection is:
Configuration config;
config.iceServers.emplace_back("stun:"+STUN_SERV);
config.enableIceUdpMux = true;
config.certificateType=rtcm::CertificateType::Rsa;
config.disableAutoNegotiation = true;
I build my project againts  -ldatachannel  -lusrsctp -ljuice -lssl -lcrypto nothin special build and linking are done
When I start the project crash occure inside method DtlsTransport::Init()  and more specially SSL_get_ex_new_index method if case DtlsTransport::TransportExIndex = -1. When DtlsTransport::TransportExIndex > 0 does not crash because did not enter in SSL_get_ex_new_index. My point is it is necessary to execute this function for the DTLS? If so please correct the issue or make some workarounds. 
10x

**Comment by pezalot** [comment]
I forgot to say am i using libwebsocket for wss to signaling:
-lwebsocket -ldatachannel -lusrsctp -ljuice -lssl -lcrypto

**Comment by pezalot** [comment]
No, my peer connection is inside a class like member var with shared_ptr

**Comment by paullouisageneau** [comment]
> No, my peer connection is inside a class like member var with shared_ptr

Sure, but is it created before the program enters main() or after?

**Comment by pezalot** [comment]
Yes, It is created after the program enters in main, this is for sure!!! No globals!!!

**Comment by paullouisageneau** [comment]
There is no specific parameter in the call to SSL_get_ex_new_index(), but it's one of the first OpenSSL functions to be called, so it's probably something wrong with OpenSSL. What is the target platform and OpenSSL version? Did you build OpenSSL yourself?
